src/core/core/api/task.py

```diff
@@ -56,3 +47,30 @@ def initialize(app: Flask):
     task_bp.add_url_rule("", view_func=Task.as_view("tasks"))
     task_bp.add_url_rule("/<string:task_id>", view_func=Task.as_view("task"))
     app.register_blueprint(task_bp)
+
+
+def serialize_result(result: Optional[dict | str] = None):
```

Comment:
to be consistent with rest of codebase use python 3.10+ type hints:

```suggestion
def serialize_result(result: dict | str | None = None):
```